src/controllers/userProfileController.js

```diff
@@ -454,7 +454,7 @@ const userProfileController = function (UserProfile, Project) {
     const isRequestorAuthorized = !!(
       canEditProtectedAccount &&
       ((await hasPermission(req.body.requestor, 'putUserProfile')) ||
-        req.body.requestor.requestorId === userid)
+        req.body.requestor.requestorId === userid || await hasPermission(req.body.requestor, 'editTeamCode'))
```

Comment:
Adding this here makes it so someone with 'editTeamCode' can edit the same fields as 'putUserProfile'. You need to add an extra check to prevent that.